std/internal/entropy/common.d

```diff
@@ -0,0 +1,181 @@
+// Written in the D programming language.
+
+/+
+    Non-backend-specific functionality.
+
+    License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+    Authors:   Elias Batek
+    Source:    $(PHOBOSSRC std/internal/entropy/common.d)
+ +/
+module std.internal.entropy.common;
+
+@nogc nothrow:
+
+///
+enum EntropySource
+{
+    /// Try supported sources one-by-one until one is available.
+    /// This exists to enable the use of this the entropy library
+    /// in a backwards compatibility way.
+    tryAll = -1,
+
+    /// Always fail.
+    none = 0,
+
+    /// `/dev/urandom`
+    charDevURandom = 1,
+
+    /// `/dev/random`
+    charDevRandom = 2,
+
+    /// `getrandom` syscall wrapper
+    getrandom = 3,
+
+    /// `arc4random`
+    arc4random = 4,
+
+    // `getentropy`
+    getentropy = 5,
+
+    /// Windows legacy
+    cryptGenRandom = 6,
+
+    /// Windows modern
+    bcryptGenRandom = 7,
+}
+
+///
+enum EntropyStatus
+{
+    ok = 0,
+    unknownError = 1,
+    unavailable,
+    unavailableLibrary,
+    unavailablePlatform,
+    readError,
+}
+
+///
+struct EntropyResult
```

Comment:
ditto

Reply:
This one now actually got a `unittest` example (that happens to test for nothing, though).